include/unifex/async_scope.hpp

```diff
@@ -170,6 +170,14 @@ struct async_scope {
     }
   }
 
+  template (typename Fun, typename Scheduler)
+    (requires callable<Fun> && scheduler<Scheduler>)
+  void spawn_with(Fun&& fun, Scheduler&& scheduler) {
+    spawn(
+      transform(schedule(), (Fun&&) fun),
+      (Scheduler&&) scheduler);
+  }
+
```

Comment:
I wonder whether this should constrain the function to be nothrow_invocable_v since it just terminates on exception/error anyway. Might be useful to require the callers to explicitly annotate their lambda/function to indicate that this is the case rather than it being hidden behaviour?
Thoughts?